packages/forma-36-react-components/src/components/NavigationIcon/NavigationIcon.tsx

```diff
@@ -0,0 +1,115 @@
+import React, { Component, CSSProperties } from 'react';
+import cn from 'classnames';
+import { navigationIconName } from './constants';
+
+import Spaces from './svg/Spaces.svg';
+import Apis from './svg/Apis.svg';
+import Billing from './svg/Billing.svg';
+import Home from './svg/Home.svg';
+import Media from './svg/Media.svg';
+import ContentModel from './svg/ContentModel.svg';
+import Content from './svg/Content.svg';
+import Apps from './svg/Apps.svg';
+import Oauth from './svg/Oauth.svg';
+import OrgInfo from './svg/OrgInfo.svg';
+import Oraganizations from './svg/Oraganizations.svg';
+import Settings from './svg/Settings.svg';
+import Sso from './svg/Sso.svg';
+import Subscription from './svg/Subscription.svg';
+import Teams from './svg/Teams.svg';
+import Token from './svg/Token.svg';
+import Usage from './svg/Usage.svg';
+import UserProfile from './svg/UserProfile.svg';
+import Users from './svg/Users.svg';
+
+import styles from './NavigationIcon.css';
+
+const iconComponents = {
+  Spaces,
+  Apis,
+  Apps,
+  Billing,
+  ContentModel,
+  Content,
+  Home,
+  Media,
+  Oauth,
+  OrgInfo,
+  Oraganizations,
+  Settings,
+  Sso,
+  Subscription,
+  Teams,
+  Token,
+  Usage,
+  UserProfile,
+  Users,
+};
+
+export type NavigationIconType = keyof typeof navigationIconName;
+export type NavigationIconColorType =
+  | 'primary'
+  | 'positive'
+  | 'negative'
+  | 'warning'
+  | 'secondary'
+  | 'muted'
+  | 'white';
+
+export type NavigationIconSize = 'small' | 'medium' | 'large' | 'xlarge';
+export type NavigationIconTagType = 'div' | 'span';
+
+export interface NavigationIconProps {
+  size?: NavigationIconSize;
+  color?: NavigationIconColorType;
+  style?: CSSProperties;
+  icon: NavigationIconType;
+  className?: string;
+  testId?: string;
+  tagType?: NavigationIconTagType
+}
+
+const defaultProps = {
+  testId: 'cf-ui-navigation-icon',
+  size: 'medium',
+  color: 'positive',
+  tagType: 'div'
+};
+
+export class NavigationIcon extends Component<NavigationIconProps & typeof defaultProps> {
```

Comment:
All good! But what do you think about making this a functional component? I think it does not need to be a class. I know may of our components are classes, but maybe we can refactor them in small steps and new components we create as functions

But maybe this is also something we should discuss as a team... Idk

Reply:
totally, I will refactor it. thx!

Reply:
@gui-santos done 🙂